### ISSUE TITLE
Export Folly strings

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/focus-android#1417]
An issue to keep track of exporting Folly strings to l10n repo.

**Comment by liuche** [comment]
Hi @pocmo, I'm running the export scripts from gradle to export strings from Folly (following the instructions [here](https://github.com/mozilla-mobile/focus-android/wiki/Localization)). I did a few additional things (run it in a virtualenv to use python 2.7.14, install the requirements.txt file from the android2po subdirectory), but I keep failing at the stringsExport step.

Any thoughts on this error? (though we're probably moving to the new version of scripts so it might not matter.)

```
:app:stringsExportTraceback

 (most recent call last):
  File "tools/l10n/android2po/a2po.py", line 1, in <module>
    from program import run
  File "/Users/liuche/code/moz/focus-android/tools/l10n/android2po/program.py", line 9, in <module>
    import argparse
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/argparse.py", line 86, in <module>
    import copy as _copy
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/copy.py", line 52, in <module>
    import weakref
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/weakref.py", line 14, in <module>
    from _weakref import (
ImportError: cannot import name _remove_dead_weakref
 FAILED

FAILURE: Build failed with an exception.
```

**Comment by pocmo** [comment]
@liuche: Try those two things:

1. In tools/l10n/android2po:
> pip install -r requirements.txt

2. Run the export script from the PR and not the gradle one. They are more reliable :)

**Comment by pocmo** [comment]
Okay, you already tried (1). Typed my previous answer from my phone and missed that. :)

**Comment by pocmo** [comment]
I recently switched to a new MacBook. I'm on Python 2.7.10 and the only thing I did is install the requirements afaik.

@mastizada You are the Python expert. Do you know what could be wrong here? :)

**Comment by liuche** [comment]
The new scripts do build for me, so that's what I'll be using for this export.

I'm seeing a lot of extraneous content (beyond the single string I've added) so I'll take a look to see what else needs to change. Also seeing #1148 too from the script, but that's probably a known problem. Will continue debugging!

**Comment by liuche** [comment]
It looks like this is a known problem, and has been around for a year:
https://github.com/miracle2k/android2po/issues/49

Since that is the case, I'm just going to make the PR and add some comments to l10n about the line-wrapping changes.

**Comment by liuche** [comment]
More notes:

There are some additional non-whitespace discrepancies in the generated content:
- Renaming of zh_TW and zh_CN in "Language" metadata field (to zh_Hant_TW and zh_Hans_CN) - https://github.com/python-babel/babel/issues/30
- The disappearance of the "X-Generator: Pontoon" metadata field from every locale

I've manually switched over the zh_CN and zh_TW fields - maybe the second issue is related, because the "X-Generator" sounds like it's some step that makes some additional changes for Pontoon.

**Comment by liuche** [comment]
PR adding new string to this repo: https://github.com/mozilla-mobile/focus-android/pull/1421
PR of exported strings l10n repo: https://github.com/mozilla-l10n/focus-android-l10n/pull/21

**Comment by mastizada** [comment]
@pocmo It was most likely problem between system's and local python versions, not related to the project at all. Sorry for late response.

**Comment by pocmo** [comment]
For reference:

> In the past we ignored metadata and whitespace/formatting changes done by the tools. On Android side they are not important and Pontoon doesn't use them either.

**Comment by liuche** [comment]
Closed in PR #1421.
